src/serde/de_br.rs

```diff
@@ -0,0 +1,144 @@
+use std::io;
+use std::io::{Cursor, Read, Seek, SeekFrom};
+
+use crate::allocator::{Allocator, NodePtr, SExp};
+use crate::traverse_path::traverse_path;
+
+use super::errors::bad_encoding;
+use super::parse_atom::{decode_size, parse_atom};
+
+const MAX_SINGLE_BYTE: u8 = 0x7f;
+const BACK_REFERENCE: u8 = 0xfe;
+const CONS_BOX_MARKER: u8 = 0xff;
+
+#[repr(u8)]
+enum ParseOp {
+    SExp,
+    Cons,
+}
+
+fn parse_path(f: &mut Cursor<&[u8]>) -> io::Result<Vec<u8>> {
```

Comment:
Oh yeah, I guess I didn't do that because I've always been a bit reticent to use `Cursor` api rather than `Read` api, as I just feel like restoring `Read` api might be useful someday. But I can cross that bridge if I come to it. I'll use your idea and make your change. I think I can actually unify the two functions too.

Reply:
Done in @a21a6a223d19d0d9923bf7fe578bac74052a9de0